addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -1867,3 +1867,35 @@ def test_check_lastcall_change_regular_to_accrual(self):
             })
 
             self.assertEqual(allocation.lastcall, datetime.date(2017, 12, 5))
+
+    def test_compute_allocation_days_after_adding_employee(self):
+        """
+        Test the addition of the employee after the date when creating an allocation
+        will the number_of_days be computed or not
+        """
+        accrual_plan = self.env['hr.leave.accrual.plan'].with_context(tracking_disable=True).create({
+            'name': 'Monthly accrual',
+            'carryover_date': 'year_start',
+            'accrued_gain_time': 'start',
+            'level_ids': [(0, 0, {
+                'added_value_type': 'day',
+                'start_count': 0,
+                'start_type': 'day',
+                'added_value': 1,
+                'frequency': 'monthly',
+                'first_day_display': '1',
+                'cap_accrued_time': False,
+            })],
+        })
+
+        with freeze_time('2024-03-02'):
+            with Form(self.env['hr.leave.allocation']) as f:
+                f.allocation_type = "accrual"
+                f.accrual_plan_id = accrual_plan
+                f.date_from = '2024-01-01'
+                f.holiday_status_id = self.leave_type
+                f.employee_ids.add(self.employee_emp)
+                f.name = "Employee Allocation"
+
+            accrual_allocation = f.record
+            self.assertNotEqual(accrual_allocation.number_of_days, 0.0, places=0)
```

Comment:
you'll need to check the number of days before and after changing the employee, using employees with different schedules to ensure that the value is updated :)

Reply:
Done :smile: